src/Board.cpp

```diff
@@ -0,0 +1,182 @@
+ Takes a string representing the initial board state and initializes the board with the pieces.
+
+ Parameters:
+ - initBoard: A string representing the initial board state. Each character in the string represents
+   a piece on the board (uppercase for white, lowercase for black), or a space for an empty square.
+   The string should be ordered from the top-left corner (A8) to the bottom-right corner (H1).*/
+Board::Board(const string& initBoard)
+    :m_turn(PieceColor::WHITE)
+{
+    for (size_t i = 0; i < initBoard.length(); ++i) {
+        auto piece = PieceFactory::createPiece(initBoard[i], convertToLocation(i));
+        if (piece) m_board.push_back(std::move(piece));
+    }
+}
+
+/*Destructor for the Board class.
+ Cleans up resources associated with the board.*/ 
+Board::~Board() {}
+
+/*Checks if the given source index is a valid index for a move.
+ The source index is valid if it corresponds to a non-empty square and the piece at that square
+ has the same color as the current player's turn.
+
+ Parameters:
+ - srcIndex: The index of the source piece in the m_board vector.
+
+ Returns:
+ - true if the source index is valid, false otherwise. */ 
+bool Board::isValidSource(int srcIndex) {
+    if (srcIndex == -1) return false;
+    return m_board[srcIndex]->get_color() == m_turn;
+}
+
+/*Checks if the given destination index is a valid index for a move.
+ The destination index is valid if it corresponds to an empty square or a square with a piece
+ of the opposite color.
+
+ Parameters:
+ - srcIndex: The index of the source piece in the m_board vector.
+ - destIndex: The index of the destination square in the m_board vector.
+
+ Returns:
+ - true if the destination index is valid, false otherwise.*/
+bool Board::isValidDestination(int srcIndex, int destIndex) {
+    return destIndex == -1 || m_board[destIndex]->get_color() != m_turn;
+}
+
+/* Checks if a move from the source index to the destination location is valid.
+ The move is valid if the piece at the source index can legally move to the destination location.
+
+ Parameters:
+ - srcIndex: The index of the source piece in the m_board vector.
+ - dest: The destination location as a Location object.
+
+ Returns:
+ - true if the move is valid, false otherwise.*/
+bool Board::isValidMove(int srcIndex, const Location& dest) {
+    return m_board[srcIndex]->isValidMove(dest);
+}
+
+/* Removes a piece from the board at the specified index.
+ If the index is -1, this function does nothing.
+
+ Parameters:
+ - destIndex: The index of the piece to remove in the m_board vector.*/
+void Board::removePiece(const int& destIndex)
+{
+    if (destIndex != -1)
+        m_board.erase(m_board.begin() + destIndex);
+}
+
+ /*Checks if the king of the current player is in check after a move.
+ This function iterates through all the opponent's pieces and checks if any of them
+ can move to the current player's king's location.
+
+ Returns:
+ - false if the king is in check, true otherwise.*/
```

Comment:
The function is called `isKingInCheckAfterMove`. From its name I would expect it to return `true` if the king is in check. Doing it the other way around confuses the reader